dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -10,346 +13,207 @@
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            DotConnect db =  new DotConnect().setSQL(SELECT);
+            return  db.loadInt("test");
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+    
+    /**
+     * Runs with a separate DB connection
+     * @return
+     */
+    private boolean createDbVersionTable() {
+
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            new DotConnect().setSQL(createTableSQL()).loadResult(conn);
+            new DotConnect().setSQL(INSERT).addParam(0).addParam(new Date()).loadResult(conn);
+            return true;
+
+        } catch (Exception e) {
+            Logger.debug(this.getClass(), e.getMessage(),e);
+            throw new DotRuntimeException(e);
+        }
+
+    }
+    
+    
+    public void executeStartUpTasks() throws DotDataException {
+        
+        Logger.debug(this.getClass(), "Running Startup Tasks");
+
+
+        String name = null;
+        try {
+            Logger.info(this, "Running Startup Tasks");
+
+
+            for (Class<?> c : TaskLocatorUtil.getStartupRunAlwaysTaskClasses()) {
+                HibernateUtil.startTransaction();
+                name = c.getCanonicalName();
+                name = name.substring(name.lastIndexOf(".") + 1);
+                if (StartupTask.class.isAssignableFrom(c)) {
+                    StartupTask  task = (StartupTask) c.newInstance();
+                    if (task.forceRun()) {
+                        HibernateUtil.startTransaction();
+                        Logger.info(this, "Running Startup Tasks : " + name);
+                        task.executeUpgrade();
+                    } else {
+                        Logger.info(this, "Not Running Startup Tasks: " + name);
+                    }
+                }
+                HibernateUtil.closeAndCommitTransaction();
+            }
+            Logger.info(this, "Finishing startup tasks.");
+        } catch (Throwable e) {
+            HibernateUtil.rollbackTransaction();
+            Logger.error(this, "FATAL: Unable to execute the upgrade task : " + name, e);
+            if(Config.getBooleanProperty("SYSTEM_EXIT_ON_STARTUP_FAILURE", true)){
+              e.printStackTrace();
+              System.exit(1);
+            }
+        } finally {
+            // This will commit the changes and close the connection
+            HibernateUtil.closeAndCommitTransaction();
+        }
+
+        
+    }
+    
+    
+    
+    
+    
+    public void executeUpgrades() throws DotDataException {
+
+        Logger.info(this, "---");
+        Logger.info(this, "");
+        Logger.info(this, "Running Upgrade Tasks");
+
+        Logger.info(this, "Database version: " + Config.DB_VERSION);
+
+        String name = null;
+
+
+        ReindexThread.pause();
+        for (Class<?> c : TaskLocatorUtil.getStartupRunOnceTaskClasses()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'c' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)